x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_modal/index.tsx

```diff
@@ -214,18 +221,29 @@ export const AddExceptionModal = memo(function AddExceptionModal({
     [setShouldBulkCloseAlert]
   );
 
+  const retrieveAlertOsTypes = useCallback(() => {
+    const osTypes = getMappedNonEcsValue({
+      data: alertData.nonEcsData,
+      fieldName: 'host.os.family',
+    });
+    if (osTypes.length === 0) {
+      return ['windows', 'macos', 'linux'];
+    }
+    return osTypes;
+  }, [alertData]);
+
   const enrichExceptionItems = useCallback(() => {
     let enriched: Array<ExceptionListItemSchema | CreateExceptionListItemSchema> = [];
     enriched =
       comment !== ''
         ? enrichExceptionItemsWithComments(exceptionItemsToAdd, [{ comment }])
         : exceptionItemsToAdd;
     if (exceptionListType === 'endpoint') {
-      const osTypes = alertData ? ['windows'] : ['windows', 'macos', 'linux'];
+      const osTypes = alertData ? retrieveAlertOsTypes() : ['windows', 'macos', 'linux'];
```

Comment:
Nit: Might be able to just clean up a bit and add the `alertData` check within `retrieveAlertOsTypes` so if someone else later uses it they don't need to remember to make this check?